libtenzir/src/tql2/parser.cpp

```diff
@@ -563,18 +564,43 @@ class parser {
     return constant{std::move(result), token.location};
   }
 
+  static auto try_double_to_integer(double x) -> constant::kind {
+    auto integral = double{};
+    auto fractional = std::modf(x, &integral);
+    if (fractional != 0.0) {
+      return x;
+    }
+    if (static_cast<double>(min<int64_t>) <= integral
+        && integral <= static_cast<double>(max<int64_t>)) {
+      return static_cast<int64_t>(x);
+    }
+    if (static_cast<double>(min<uint64_t>) <= integral
+        && integral <= static_cast<double>(max<uint64_t>)) {
+      return static_cast<uint64_t>(x);
+    }
```

Comment:
```suggestion
    if (static_cast<double>(std::numeric_limits<int64_t>::min()) <= integral
        && integral <= static_cast<double>(std::numeric_limits<int64_t>::max())) {
      return static_cast<int64_t>(x);
    }
    if (static_cast<double>(std::numeric_limits<uint64_t>::min()) <= integral
        && integral <= static_cast<double>(std::numeric_limits<uint64_t>::max())) {
      return static_cast<uint64_t>(x);
    }
```

Doesn't compile for me without this change.

Reply:
Missed an include. We have aliases for those (not only because it's easier to type, but also because `::min` and `::lowest` is just a wild design).